MaterialBudget.cpp

```diff
@@ -113,4 +125,44 @@ Particle* MaterialBudget::MultScattering(Particle* part)
     return part;
 }
 
+MaterialBudget::fPoint MaterialBudget::GetIntersection(const Particle* particle, bool)
```

Comment:
Probably a better implementation could be setting the function to void and passing a pointer to a MaterialBudget::fPoint as a parmeter (instead of the bool), if the pointer is not nullptr, then set the object pointed by pointer to be equal to the intersection